plugins/machina-tools.js

```diff
@@ -0,0 +1,654 @@
+  var i = size === 0 ? 0 : Math.floor(Math.log(size) / Math.log(1024));
+  return (size / Math.pow(1024, i)).toFixed(2) * 1 + ' ' + ['B', 'kiB', 'MiB', 'GiB', 'TiB'][i];
+}
+
+var typeSizes = {
+  undefined: () => 0,
+  boolean: () => 4,
+  number: () => 8,
+  string: (item) => 2 * item.length,
+  object: (item) => (!item ? 0 : Object.keys(item).reduce((total, key) => (Array.isArray(item) ? 0 : sizeOf(key)) + sizeOf(item[key]) + total, 0)),
+};
+
+function sizeOf(value) {
+  return typeSizes[typeof value](value);
+}
+
+function createAreaInfoDialogContent() {
+  var html = $('<div>');
+  if (machinaTools.conflictAreaLast && machinaTools.conflictAreaLast.getLayers().length > 0) {
+    var ul = $('<ul>');
+    ul.appendTo(html);
+
+    var conflictAreaLayers = machinaTools.conflictAreaLast.getLayers();
+    let conflictLayer = conflictAreaLayers[0];
+    ul.append($('<li>', { html: `Cluster area size: ${humanFileSize(sizeOf(conflictLayer.feature))}` }));
+
+    var unitedAreas = 0;
+    var totalPoints = 0;
+
+    var conflictLayerGeometry = conflictLayer.feature.geometry;
+    if (conflictLayerGeometry.type === 'Polygon') {
+      unitedAreas = 1;
+      totalPoints = conflictLayerGeometry.coordinates[0].length;
+    } else if (conflictLayerGeometry.type === 'MultiPolygon') {
+      unitedAreas = conflictLayerGeometry.coordinates.length;
+      totalPoints = conflictLayerGeometry.coordinates.flatMap((v) => v).flatMap((v) => v).length;
+    }
+    ul.append($('<li>', { html: `Continuous areas: ${unitedAreas}` }));
+    ul.append($('<li>', { html: `Total area points: ${totalPoints}` }));
+  } else {
+    html.append('No machina clusters found');
+  }
+  return html;
+}
+
+function refreshDialogs(guid) {
+  if (guid && window.portals[guid] && window.TEAM_NAMES[window.portals[guid].options.team] === window.TEAM_NAME_MAC) {
+    var seed = machinaTools.findSeed(guid);
+    if (seed && machinaTools._clusterDialogs[seed.guid]) {
+      doDisplayClusterInfo(seed);
+    }
+  }
+  if (machinaTools._conflictAreaInfoDialog) {
+    machinaTools._conflictAreaInfoDialog.html(createAreaInfoDialogContent());
+  }
+}
+
+machinaTools.showConflictAreaInfoDialog = function () {
+  machinaTools._conflictAreaInfoDialog = dialog({
+    html: createAreaInfoDialogContent(),
+    title: 'Machina Conflict Area Info',
+    id: 'machina-conflict-area-info',
+    width: 'auto',
+    closeCallback: () => {
+      delete machinaTools._conflictAreaInfoDialog;
+    },
+    buttons: {
+      'Reset Conflict Area': machinaTools.resetConflictArea,
+    },
+  });
+};
+
+machinaTools.loadConflictAreas = function () {
+  Object.values(window.portals)
+    .filter((p) => window.TEAM_NAMES[p.options.team] === window.TEAM_NAME_MAC)
+    .forEach((portal) => machinaTools.drawPortalExclusion(portal));
+
+  Object.values(window.links)
+    .filter((l) => window.TEAM_NAMES[l.options.team] === window.TEAM_NAME_MAC)
+    .forEach((link) => machinaTools.drawLinkExclusion(link.options.data));
```

Comment:
idea: since `machinaTools.drawPortalExclusion` takes a portal as argument, the `drawLinkExclusion` could take a link instead of link data.

 ```js
 // ...
   .forEach(machinaTools.drawPortalExclusion);
// ...
   .forEach(machinaTools.drawLinkExclusion);
```

Reply:
Valid comment, but in the end we are discussing about an experimental plugin. IMHO this would be different, if this was a core function or we'd have an idea to export that function to other plugins.

Reply:
this is about code quality/consistency
while you say "experimental", the word "experimental" is nowhere to be found in the code, the experimental folder could be used, but it looks plugins there are not part of the releases

Reply:
i agree, it's more about code quality also, any PR should be a "closed" or "finished" feature meaning that we won't need to update the code unless we'll really need or want to - changes in 3rd party apis, new features, etc, but not to "finish tests later".
It might seem a bit tidious... and it sometimes really is, but it makes the code better understandable and extendable so even if we spend some more time now on this, we'll save much more time cumulatively moving forward.
Even more so with 'experimental' features because we know we'll want to update it quite soon, so why make it harder for ouselfs 😉

Reply:
aand done